presto-main/src/main/java/com/facebook/presto/cost/VariableStatsEstimate.java

```diff
@@ -0,0 +1,245 @@
+
+        checkArgument(averageRowSize >= 0 || isNaN(averageRowSize), "Average row size should be non-negative or NaN, got: %s", averageRowSize);
+        this.averageRowSize = averageRowSize;
+
+        checkArgument(distinctValuesCount >= 0 || isNaN(distinctValuesCount), "Distinct values count should be non-negative, got: %s", distinctValuesCount);
+        // TODO normalize distinctValuesCount for an empty range (or validate it is already normalized)
+        this.distinctValuesCount = distinctValuesCount;
+    }
+
+    @JsonProperty
+    public double getLowValue()
+    {
+        return lowValue;
+    }
+
+    @JsonProperty
+    public double getHighValue()
+    {
+        return highValue;
+    }
+
+    @JsonProperty
+    public double getNullsFraction()
+    {
+        return nullsFraction;
+    }
+
+    public StatisticRange statisticRange()
+    {
+        return new StatisticRange(lowValue, highValue, distinctValuesCount);
+    }
+
+    public double getValuesFraction()
+    {
+        return 1.0 - nullsFraction;
+    }
+
+    @JsonProperty
+    public double getAverageRowSize()
+    {
+        return averageRowSize;
+    }
+
+    @JsonProperty
+    public double getDistinctValuesCount()
+    {
+        return distinctValuesCount;
+    }
+
+    public VariableStatsEstimate mapNullsFraction(Function<Double, Double> mappingFunction)
+    {
+        return buildFrom(this).setNullsFraction(mappingFunction.apply(nullsFraction)).build();
+    }
+
+    public VariableStatsEstimate mapDistinctValuesCount(Function<Double, Double> mappingFunction)
+    {
+        return buildFrom(this).setDistinctValuesCount(mappingFunction.apply(distinctValuesCount)).build();
+    }
+
+    public boolean isUnknown()
+    {
+        return this.equals(UNKNOWN);
+    }
+
+    public boolean isSingleValue()
+    {
+        return distinctValuesCount == 1.0
+                && Double.compare(lowValue, highValue) == 0
+                && !isInfinite(lowValue);
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        VariableStatsEstimate that = (VariableStatsEstimate) o;
+        return Double.compare(nullsFraction, that.nullsFraction) == 0 &&
+                Double.compare(averageRowSize, that.averageRowSize) == 0 &&
+                Double.compare(distinctValuesCount, that.distinctValuesCount) == 0 &&
+                Double.compare(lowValue, that.lowValue) == 0 &&
+                Double.compare(highValue, that.highValue) == 0;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(lowValue, highValue, nullsFraction, averageRowSize, distinctValuesCount);
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("range", format("[%s-%s]", lowValue, highValue))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format double: lowValue, highValue. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 173. Consider formatting this data with &#39;%f&#39; instead.